code/modules/antagonists/cult/runes.dm

```diff
@@ -387,6 +387,8 @@ structure_check() searches for nearby cultist structures required for the invoca
 
 /obj/effect/rune/teleport/Destroy()
 	LAZYREMOVE(GLOB.teleport_runes, src)
+	QDEL_NULL(inner_portal)
+	QDEL_NULL(outer_portal)
```

Comment:
What happens if the portal is deleted before the rune?

Reply:
That somehow did not pop up during testing, thanks

Reply:
it's alright to call qdel() on null, my question is more, if a portal is deleted separately from the rune, will this reference be cleared?

Reply:
Oh that's why it didn't show up. So the portals are /obj/effect/temp_visual's, which are indestructible. They delete themselves when the time passes (in this case, 600 deciseconds), or when code explicitly deletes them. In this case, close_portal() acts to clean up the references, as its called either when 600 deciseconds have passed, or when the old portal needs to be deleted explicitly and a new one made. The reason references weren't cleaning up were because prior to this, if 600 deciseconds passed and the portal got deleted that way, inner_portal and outer_portal weren't reset to null.